### PR TITLE
Add displayName param of adapter

### PR DESCRIPTION
**Pull request by PorterNan**
- Add a required displayName parameter for adapter
- Add an optional displayName of StatefulClient for CTE
- Update the sample code to support it in sample

# What

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/calling-stateful-client/src/StatefulCallClient.ts [hunk 1]
```diff
@@ -193,7 +193,9 @@ class ProxyCallClient implements ProxyHandler<CallClient> {
             const callAgent = await target.createCallAgent(...args);
             this._callAgent = callAgentDeclaratify(callAgent, this._context, this._internalContext);
```
Comment (alkwa-msft) [inline packages/calling-stateful-client/src/StatefulCallClient.ts:194]: is it possible for teamsCallAgent to also take in the displayname? we take in a displayname in two different ways based on if its teams or acs?

Comment (alkwa-msft) [reply packages/calling-stateful-client/src/StatefulCallClient.ts:194]: I suppose its ok if we are only going into public preview as we could change this in the future. Its just odd that we have two different ways of setting display name if your using acs or teams

File: samples/Calling/src/app/views/HomeScreen.tsx [hunk 2]
```diff
@@ -263,23 +263,21 @@ export const HomeScreen = (props: HomeScreenProps): JSX.Element => {
...
             onClick={() => {
-              if (displayName || /* @conditional-compile-remove(teams-identity-support) */ teamsIdentityChosen) {
+              if (displayName) {
                 displayName && saveDisplayNameToLocalStorage(displayName);
                 /* @conditional-compile-remove(one-to-n-calling) */
                 const acsParticipantsToCall = parseParticipants(outboundParticipants);
                 /* @conditional-compile-remove(PSTN-calls) */
                 const dialpadParticipantToCall = parseParticipants(dialPadParticipant);
                 props.startCallHandler({
                   //TODO: This needs to be updated after we change arg types of TeamsCall
```
Comment (dmceachernmsft) [inline samples/Calling/src/app/views/HomeScreen.tsx:279]: is this TODO still something we need to worry about?

**Review by dmceachernmsft — APPROVED** [review]

File: packages/communication-react/review/beta/communication-react.api.md [hunk 3]
```diff
@@ -1664,7 +1665,7 @@ export const createStatefulCallClient: (args: StatefulCallClientArgs, options?:
 export const createStatefulChatClient: (args: StatefulChatClientArgs, options?: StatefulChatClientOptions | undefined) => StatefulChatClient;
 
 // @beta (undocumented)
-export const createTeamsCallAdapter: ({ userId, credential, locator }: TeamsCallAdapterArgs) => Promise<TeamsCallAdapter>;
+export const createTeamsCallAdapter: ({ userId, credential, displayName, locator }: TeamsCallAdapterArgs) => Promise<TeamsCallAdapter>;
```
Comment (JamesBurnside) [inline packages/communication-react/review/beta/communication-react.api.md:1668]: from discussion that CTE <--> CTE callees won't see each others' names do we still want to go ahead with this change?

**Comment by PorterNan** [comment]
Close this one because we are seeking for no displayname by default + displayName override

**Pull request closed without merging**